### ISSUE TITLE
Roadmap & stable release

### ISSUE THREAD

**Issue by rctec-at** [adospace/reactorui-maui#14]
Is there a official Roadmap and maybe a planed ETA for a stable release of this package?

**Comment by adospace** [comment]
Hi, 
sorry for the late reply, I'm planning to move the library from BETA to RELEASE stage before the end of this month (Jan-2023).

The library itself is already stable enough to be used in a production application, I'm planning to polish the code a bit removing old stuff not used anymore before releasing it.

To be honest, I hoped to have more feedback from adopters regarding problems or issues but I guess the user base is still small.

I think anyway it will gain some relevance as I officially declare it GA.

Thanks for your interest!

**Comment by LITTOMA** [comment]
amazing.

**Comment by adospace** [comment]
Happy to notify watchers of this ticket that MauiReactor is GA since version 1.0.105.
I'm going to fix bugs and add new features without changing anything on the API surface so that the numerous samples provide will always build without errors.

Regarding documentation please take a look at the official MauiReactor doc site: https://adospace.gitbook.io/mauireactor/

I'll keep this ticket open for further collecting info about the release & doc that is surely helpful to others too.

**Comment by johnkattenhorn** [comment]
This looks really interesting @adospace - thanks for all the hard work so far! I watched the video - do you have a rough timescale on when the testing solution you mentioned in the video will be available to try out?

**Comment by adospace** [comment]
Thanks for your interest @johnkattenhorn, yes I was envisioning building a testing framework for MauiReactor and it's something to do in the coming months, I'll keep you updated on my progress on this thread.
Said that, as you know, testing MAUI applications itself is not so straightforward because we'll end up building more integration tests than unit tests: the reason is that we need to actually create MAUI/Native controls under the hood.

On this note, I'm sure you know about the Spice framework https://github.com/jonathanpeppers/spice from @johnkattenhorn: that would be probably a game changer for MauiReactor because it's actually what my library really needs (i.e. a minimal framework to just build native controls tree without the overhead of the MVVM stack). 
That framework (if it will become real one day) will allow us to build real unit tests without dotnet MAUI dependencies.
I've even built a MauiReactor prototype based on Spice (https://github.com/adospace/spice) that seems working pretty well.

So to sum up, I'm still on track to release a testing framework for MauiReactor in the coming months but thinking about the future I guess we still need better support for testing from MAUI itself.

**Comment by johnkattenhorn** [comment]
Thanks, @adospace I'm going to take a look in more detail over the weekend, we are pretty passionate about the possibilities of doing UI via code; you might know our technical director, who is also working in this space too (https://github.com/VincentH-Net/CSharpForMarkup#readme)

**Comment by timahrentlov** [comment]
I love the concept! Great work. And great samples.

Being so close to how flutter works, what are your thoughts about an AvaloniaUI 11 version?

**Comment by adospace** [comment]
Thanks @timahrentlov, 

I follow Avalonia development closely, I make experiments with it since I discovered the project a few years ago. Besides the numerous great features, when I develop with it I still miss a lot of the MVU pattern and boost of productivity that comes with it from a real hot reload function.

Said that I always found that's beneficial for the community to have so many competing great alternatives to building UI in dotnet considering also Uno and Blazor.

**Comment by timahrentlov** [comment]
Thanks you for sharing your sentiments. It is much appreciated!

**Comment by adospace** [comment]
Wanted just to let you know that the latest version of MauiReactor fully supports component testing, 
Please check some sample tests here: https://github.com/adospace/reactorui-maui/tree/main/samples/UnitTests
Also for the documentation, there is a new article: https://adospace.gitbook.io/mauireactor/components/testing

@johnkattenhorn

**Comment by johnkattenhorn** [comment]
> Wanted just to let you know that the latest version of MauiReactor fully supports component testing, Please check some sample tests here: https://github.com/adospace/reactorui-maui/tree/main/samples/UnitTests Also for the documentation, there is a new article: https://adospace.gitbook.io/mauireactor/components/testing
> 
> @johnkattenhorn

- This is awesome news! @adospace - I'm checking it out now and I'll send you some feedback!

**Comment by adospace** [comment]
Hi everyone, with latest PR, MauiReactor now supports also builtin dotnet maui hot-reload function (MetadataUpdateHandlerAttribute).
A new documentation page is coming

**Comment by stillnurs** [comment]
> Hi everyone, with latest PR, MauiReactor now supports also builtin dotnet maui hot-reload function (MetadataUpdateHandlerAttribute). A new documentation page is coming

Hi, thanks for your great work.
Does this mean that we don't need to run Reactor.Maui.HotReload anymore?

**Comment by adospace** [comment]
@stillnurs no it's just another way to hot-reload the application. The custom MauiReactor hot-reload still supports much more code modifications compared to the built-in one.
I guess that the dotnet built-in hot-reload will be more and more powerful in the future (see .NET 8+)
Updated list of supported edits in dotnet: https://github.com/dotnet/roslyn/blob/main/docs/wiki/EnC-Supported-Edits.md

**Comment by stillnurs** [comment]
@adospace Is there any point to combine MauiReactor with C# Markup Extensions from Community Toolkit? 
I'm fairly new to .NET Maui, hope I've given a proper question.

**Comment by saint4eva** [comment]
@stillnurs  They use different patterns. MauiReactor uses C# MVU app development pattern, while C# Markup Extension is heavily influenced by MVVM. So, they adhere to distinctive design patterns.

**Comment by Otuyishime** [comment]
@adospace I apologize I couldn't find anywhere else to ask you this question, but Is there a way to smoothly reverse animations with AnimationController? I've playing with it and it seems toggling _IsEnabled_ resets controls to their original position. 
You can use this small component as an example. Appreciated.
```
class AnimatedBoxState
{
    public double Translate { get; set; } = 0.0;
    public bool ToggleAnimation { get; set; } = false;
}

class AnimatedBox : Component<AnimatedBoxState>
{
    public override VisualNode Render()
    {
        return new ContentPage("Animated Box")
        {
            new Grid()
            {
                new AnimationController
                {
                    new SequenceAnimation
                    {
                        new DoubleAnimation()
                            .StartValue(0)
                            .TargetValue(50)
                            .Duration(TimeSpan.FromMilliseconds(600))
                            .Easing(Easing.SinInOut)
                            .OnTick(v => SetState(s => s.Translate = v)),
                    }
                    .Loop(false)
                }
                .IsEnabled(State.ToggleAnimation)
                .OnIsEnabledChanged(enabled => SetState(s => s.ToggleAnimation = enabled)),
                new BoxView()
                    .GridRow(0)
                    .HCenter()
                    .VCenter()
                    .HeightRequest(80)
                    .WidthRequest(80)
                    .CornerRadius(10)
                    .TranslationX(State.Translate),
                new Button("Animate")
                    .GridRow(1)
                    .HCenter()
                    .VCenter()
                    .WidthRequest(150)
                    .OnClicked(Toggle)
            }
            .Rows(new MauiControls.RowDefinitionCollection
            {
                new MauiControls.RowDefinition(),
                new MauiControls.RowDefinition(GridLength.Auto)
            })
            .Padding(10)
            .BackgroundColor(Colors.LightSkyBlue)
        };
    }

    private void Toggle()
    {
        SetState(s => s.ToggleAnimation = !s.ToggleAnimation);
    }
}
```

**Comment by adospace** [comment]
@Otuyishime replied here: https://github.com/adospace/reactorui-maui/issues/59

**Comment by adospace** [comment]
Working on a new feature: it will be possible to use MauiReactor only for a portion of your XAML-based standard MVVM application using a new control called ComponentHost.

You'll be able to use MauiReactor in multiple locations or for a whole Page. Hot-reload will work too (of course only for MauiReactor components).

Everything seems to work well, so I will release it very soon, but if you want to experiment with it now, please pull the xaml-integration branch.

https://github.com/adospace/reactorui-maui/assets/10573253/f950f09f-0429-4d4f-80da-f140776f7b29

**Comment by freever** [comment]
What a fantastic idea!

**Comment by saint4eva** [comment]
That i

> Working on a new feature: it will be possible to use MauiReactor only for a portion of your XAML-based standard MVVM application using a new control called ComponentHost.
> 
> You'll be able to use MauiReactor in multiple locations or for a whole Page. Hot-reload will work too (of course only for MauiReactor components).
> 
> Everything seems to work well, so I will release it very soon, but if you want to experiment with it now, please pull the xaml-integration branch.
> 
>  2023-05-23.17-35-23.mp4

That is a nice effort to encourage people to progressive migrate their app to Maui Reactor. But do not put so much effort on it as it might still encourage people to stick to MVVM and Xaml world. 

The purpose of Maui Reactor is MVU and C#

**Comment by acaliaro** [comment]
@adospace ti dicono anche quello che devi o non devi fare, fantastico. Buon lavoro

**Comment by iamnevir** [comment]
i have problem how to get api for maui reactor, do you have any example of it?

**Comment by adospace** [comment]
@iamnevir sorry, not sure what you need, do you mean the documentation:
https://adospace.gitbook.io/mauireactor/

**Comment by iamnevir** [comment]
have any sample app of mauireactor work with api? i need some to learn

**Comment by adospace** [comment]
I have a sample app that I'm crafting that interacts with a REST API including Signalr, but it is not yet ready, I'll let you know here about my progress.
This is the sample app (not yet working):
https://github.com/adospace/mauireactor-samples/tree/main/ChateoApp

**Comment by iamnevir** [comment]
![image](https://github.com/adospace/reactorui-maui/assets/128579399/21a1608a-1b8b-4101-88e0-fa40a8b40702)

how can i make this in maui? pls help

**Comment by adospace** [comment]
@iamnevir please open a new issue for that, I use this thread only for announcements

**Comment by adospace** [comment]
Hi, MauiReactor is built on top of .NET Maui so generally speaking it supports any control that works with MAUI itself, so including DevExpress, SyncFusion, Grial, Telerik etc.

Said that, the integration can be sometimes tricky, especially for complex controls like CollectionViews, DataGrids etc.
The Scaffold attribute tries to generate a general-purpose wrapper that usually works as is, but sometimes it requires further modifications and improvements.

Even if I personally think that in many cases using special libraries is not required to create even complex UIs I recognize that this can be a problem for MauiReactor adoption.

For this reason, I decided to create a repository devoted to containing and sharing wrappers for well-known libraries.

[mauireactor-integration repo](https://github.com/adospace/mauireactor-integration)

The first wrapper I'm going to add is for DevExpress DxCollectionView along with a sample app.

Please create an issue on that repo if you find need advice on third-party control library integrations.

**Comment by AswinPG** [comment]
> Working on a new feature: it will be possible to use MauiReactor only for a portion of your XAML-based standard MVVM application using a new control called ComponentHost.
> 
> You'll be able to use MauiReactor in multiple locations or for a whole Page. Hot-reload will work too (of course only for MauiReactor components).
> 
> Everything seems to work well, so I will release it very soon, but if you want to experiment with it now, please pull the xaml-integration branch.
> 
>  2023-05-23.17-35-23.mp4

Hi @adospace Is there documentation for this feature. I would love to try it out. :)

**Comment by adospace** [comment]
The feature is still WIP and to be honest I'm not sure whether to publish it or not ( :) ).
Anyway, if you want to check it out, please download this branch:
https://github.com/adospace/reactorui-maui/tree/xaml-integration

thee you can find a sample using the xaml-maui.reactor feature

**Comment by AswinPG** [comment]
> The feature is still WIP and to be honest I'm not sure whether to publish it or not ( :) ). Anyway, if you want to check it out, please download this branch: https://github.com/adospace/reactorui-maui/tree/xaml-integration
> 
> thee you can find a sample using the xaml-maui.reactor feature

Thanks I'll check if out.
Any reason why you're not sure to publish it? It'll definitely help more people come to maui reactor easily

**Comment by adospace** [comment]
Mixing MVU and MVVM in a single app could be somewhat confusing: it can be helpful in some cases but certainly could lead to a disaster if not done correctly.

**Comment by juanyacovino77** [comment]
why not just to define and to pass parameters (aka component props") to OnMount() methods
 or maybe using the constructor insted of using State mechanism?

**Comment by adospace** [comment]
Starting with version 1.0.138, MauiReactor allows the integration of components in standard XAML or C# based applications using the `ComponentHost` control.

For more info please take a look at the documentation: https://adospace.gitbook.io/mauireactor/components/xaml-integration

Enjoy!

**Comment by juanyacovino77** [comment]
> Starting with version 1.0.138, MauiReactor allows the integration of components in standard XAML or C# based applications using the `ComponentHost` control.
> 
> For more info please take a look at the documentation: https://adospace.gitbook.io/mauireactor/components/xaml-integration
> 
> Enjoy!

Would this allow us to use MauiReactor components inside an UNO PLATFORM app?

**Comment by juanyacovino77** [comment]
Hey, 

How is FlyoutPage supposed to be used at MauiReactor? 

There are not FlyoutPage.Flyout nor FlyoutPage.Details properties registered neither SelectionChanged events for listening to selection of a FlyoutPageItem to display details of it..   


Thankss

**Comment by adospace** [comment]
@juanyacovino77 have you seen the documentation: https://adospace.gitbook.io/mauireactor/components/navigation/shell ?

please open new tickets for any issues you may encounter, I'm using the ticket for announcements and general comments, thanks

**Comment by juanyacovino77** [comment]
> @juanyacovino77 have you seen the documentation: https://adospace.gitbook.io/mauireactor/components/navigation/shell ?
> 
> please open new tickets for any issues you may encounter, I'm using the ticket for announcements and general comments, thanks

Yes, I 've seen it. But I want to use FlyoutPage not to use FlyoutItem within a Shell app. Is it possible to just use FlyoutPage without shell app at maui reactor?

**Comment by adospace** [comment]
@juanyacovino77 here there is an example: https://github.com/adospace/kee-mind/blob/main/src/KeeMind/Pages/MainPage.cs

**Comment by Ohior** [comment]
I just discovered MauiReactor, XAML Looks scary, 
I have been testing it for some days now before I will use it for production. I can't seem to find a way to navigate from one page to another and share data between them. I am new to both C# and XAML, I prefer MauiReactor because unlike flutter and compose you can test your app on IPhone devices when developing on windows,

**Comment by tcsaddul** [comment]
Maybe you should first look at the Sample Apps [https://github.com/adospace/mauireactor-samples]. The Chateo app has some pages in it.

**Comment by Ohior** [comment]
> @Ohior please take a look at the documentation: https://adospace.gitbook.io/mauireactor/components/navigation

Thanks, This is a lot easy to understand and implement

**Comment by Ohior** [comment]
I have read the document on [ implementing popup](https://adospace.gitbook.io/mauireactor/q-and-a/how-to-deal-with-custom-dialogs-popups) and it is tedious to implement. Is there  a simpler way.  I still have not figured it out.

Also there are so many namespace conflicts, this makes me spend a lot of time debugging. For someone new this might be a big issue.

**Comment by Code-DJ** [comment]
Hi @adospace, just FYI - I did an upgrade to .net 8 rc1 to see how well it works. As soon as I ran the solution, found that button click didn't work. Is it because of Maui.Reactor's dependency on .net 7? Not planning on upgrade until GA, was just curious of the dependencies. Thanks!

**Comment by raselldev** [comment]
Hi @adospace current version is stable release?

**Comment by adospace** [comment]
Yes it's a stable release, please always pick latest from nuget

**Comment by cris-m** [comment]
I would like to know if there is a ways to use `Behavior`. Like to change color of `svg` image using `IconTintColorBehavior`  of `.NET MAUI Community Toolkit`

**Comment by adospace** [comment]
@cris-m please open a specific ticket describing the issue in detail, thanks

**Comment by liwoo** [comment]
Hi @adospace Awesome Work 👏🏾 on ReactorUI.  I've been playing around with it for the last few weeks, and even though I have had to do a lot of digging into source code to find exactly what I need (for example - how to tint Toolbar text with a different color from the ContentPage title - as is the default in iOS), I've been pleasantly surprised by the sheer feature set and quality of this library!

I just wanted to find out what your thoughts are around having a globally applicable data management solution, something like SwiftData.  Do you think one easily maintain state in a complex app (say 100s of pages) using OnMount and SetState?  

I've used the Bloc Pattern in Flutter in the past and I find it easier to reason about complex apps.  Thoughts?

**Comment by adospace** [comment]
Hi @liwoo, I'm glad you're considering MauiReactor for your apps.

Let me answer the state manager thing first and let's consider a large app only (for a small app with a few pages I don't think is a good idea to use a different state manager other than the one provided by the framework).
 
I'm not a huge expert on state managers but I made more than an experiment in .net app MauiReactor and Blazor using Fluxor and Memento. 
MauiReactor has its internal "state manager", which is minimal that serves only the purpose of the library. Someone could even complain that I call it state manager, anyway it works and it's easy to use.
This is a possible integration of Flux for MauiReactor:
https://gist.github.com/adospace/cbede42c410c642dbdbcafe9ece5e90b

Someone could say that state manager was invented for those languages that didn't have a proper DI container with the ability to inject services easily.

State managers can help to maintain discipline inside the project and have some advantages but really what is annoying is the complexity it adds to it.

So the real benefit comes IMHO when you have a large project + many (5+) developers on the team.

Instead, I can say that I officially fell in love with SwiftData (that I didn't know before you asked the question). 
I so liked it that I created my library ReactorData designed after SwiftData and so far I must say that it's pleasing to work with.

@all guys, I would love to have your opinions about **ReactorData**!
@liwoo please check it out
https://github.com/adospace/reactor-data

this is what could be a todo app using MauiReactor+ReactorData
https://github.com/adospace/mauireactor-samples/blob/main/TodoApp/Pages/MainPage.cs

```csharp
class MainPageState
{
    public IQuery<Todo> TodoItems { get; set; } = default!;
}

partial class MainPage : Component<MainPageState>
{
    [Inject]
    IModelContext _modelContext;

    protected override void OnMounted()
    {
        State.TodoItems = _modelContext.Query<Todo>(query => query.OrderBy(_ => _.Task));

        base.OnMounted();
    }

    public override VisualNode Render()
        => ContentPage(
            Grid("Auto, *, Auto", "*",
                TodoEditor(OnCreatedNewTask),

                CollectionView()
                    .ItemsSource(State.TodoItems, RenderItem)
                    .GridRow(1),

                Button("Clear List")
                    .OnClicked(OnClearList)
                    .GridRow(2)

            ));

    

    VisualNode RenderItem(Todo item)
        => Grid("54", "Auto, *",
            CheckBox()
                .IsChecked(item.Done)
                .OnCheckedChanged((s, args) => OnItemDoneChanged(item, args.Value)),
            Label(item.Task)
                .TextDecorations(item.Done ? TextDecorations.Strikethrough : TextDecorations.None)
                .VCenter()
                .GridColumn(1));

    static VisualNode TodoEditor(Action<Todo> created)
        => Render<string>(state =>
            Grid("*", "*,Auto",
                Entry()
                    .Text(state.Value ?? string.Empty)
                    .OnTextChanged(text => state.Set(s => text, false)),
                Button("Create")
                    .GridColumn(1)
                    .OnClicked(() =>
                    {
                        created(new Todo { Task = state.Value ?? "New Task" });
                        state.Set(s => string.Empty);
                    })
                )
            );

    void OnItemDoneChanged(Todo item, bool done)
    {
        item.Done = done;

        _modelContext.Update(item);
        _modelContext.Save();
    }

    void OnCreatedNewTask(Todo todo)
    {
        _modelContext.Add(todo);
        _modelContext.Save();
    }

    void OnClearList()
    {
        _modelContext.DeleteRange(State.TodoItems);
        _modelContext.Save();
    }
}
```

**Comment by adospace** [comment]
New feature alert!  :)

I'm working to add the ability to style widgets in an easier and more centralized way.

Currently theming is provided directly from the .NET MAUI framework using styles defined either in XAML or in code.

In addition to the standard method (that, of course, remains and will be fully supported in the future), there will be another way to style widgets.

MauiReactor will provide a new class for each widget called after its name adding the "Styles" post-fix:.
For example, you'll find LabelStyles, ButtonStyles, ViewStyles etc.

To change the default style of a widget you just have to modify its Default style in the relative static class.

for example:

```c#
LabelStyles.Default = _ => _.FontSize(28).FontFamily("MyFont");
ButtonStyles.Default = _=> _.Padding(5);
```
You'll be able to modify also the style of more generic classes, for example, to change the margin of all derived View classes you can simply set this style:

`ViewStyles.Default = _=> _.Margin(2);`

Furthermore, it will be possible to define specific selectors (like the classes in CSS) that will allow you to specify different style for the same widget.
For example:

```c#
LabelStyles.Themes["H1"] = _=>_.FontSize(18);
LabelStyles.Themes["SubTitle"] = _=>_.FontSize(16);
```

and you can select the style you need in the component using the Theme property, like it's shown below:

```c#
VStack(
   Label("Title").Theme("H1),
   Label("SubTitle").Theme("SubTitle")
)
```

If you want to experiment with this new function please attach the "theming-feature" branch (https://github.com/adospace/reactorui-maui/tree/theming-feature).

Check it out!

**Comment by juanyacovino77** [comment]
Wouldn't be possible to use the same source generator mechanism of [Prop], [Param], and [Inject] attributes but with State properties declared inside the class?

**Comment by adospace** [comment]
@juanyacovino77 it's something I'm looking into...

**Comment by adospace** [comment]
The theming feature now supports Dark/Light theme, MauiReactor responds automatically to theme change requests coming from the user or OS and applies the theme accordnly:

https://adospace.gitbook.io/mauireactor/components/theming#dark-theme-support 

![DarkThemeSupport](https://github.com/adospace/reactorui-maui/assets/10573253/22109bfb-8164-4e3f-a0aa-1211aacd401d)

**Comment by ThaDaVos** [comment]
Just a question out of curiosity - but how does this differ from [Comet](https://github.com/dotnet/Comet) which also aims for bringing MVU to MAUI?

**Comment by adospace** [comment]
@ThaDaVos 
MauiReactor is a full MVU framework, it works mostly like ReactJS (and Flutter). 
Consider it like this:
ReactJS creates components over the HTML DOM, when the render is called a new visual tree is created, and it is compared to the old one. New HTML nodes are created, old ones are removed, and existing ones are updated.
MauiReactor creates components over the MAUI.NET tree of controls, and mostly the same happens: when the render is called, a new visual tree is called and compared to the old one. New maui.net controls are created, old ones are removed, and existing ones are updated.

Flutter works differently using Skia to render widgets but the core idea is similar. I was inspired by ReactJs anyway.

In MauiReactor you have views and models, views are rendered every time you call SetState() or Invalidate (similar to ReactJS). Models contain the state that is preserved between renders.
Decoupling state and views make hot-reload work very well: I rarely need to restart the app when working.

More info here https://adospace.gitbook.io/mauireactor

Comet builds views more like c# markup (so in pure c#) and then creates direct binding with the state class that automatically implements Observable properties. 
It directly creates native controls (so it doesn't use virtual trees) through MAUI custom handlers.
I can't say if it can be considered an MVU pure approach, but neither MauiReactor is.
Comet seems inspired by SwiftUI.

What I like about Comet:
- It doesn't need MAUI controls at all just the core framework required to create handlers
- It's fast as it directly binds the state to the properties of the controls

What I do not like:
- States and views are coupled, so things like hot-reload are much more hard to implement
- Conditional rendering offers the same problems as the MVVM approach
- You can't directly integrate third-party controls from open-source libraries or vendors (Telerik, etc)

I never used Comet so take what I say as is.

I even less know MVUX from Uno, so I'm not sure how it compares with MauiReactor (MauiReactor came before it).

**Comment by adospace** [comment]
Hi @ziaulhasanhamim, regarding performance I can say that is pretty much the same as the underlying framework (MAUI). Yes, MauiReactor doesn't use reflection so it's fully trimmable and AOT-able.
I think MauiReactor is a bit faster to display pages on open because actually, it doesn't need to connect bindings on viewmodels (of course if you do not use bindings/MVVM or only compiled bindings then is the same).

Generally speaking, most of the performance issues I encountered in the past were caused by mistakes I made in code that were not related to the platform (and I guess the same happens with standard XAML applications).

Regarding the second question for "simple linq" do you mean just iterate over a collection and create items?

```c#
VScrollView(
  VStack(
    State.Items.Select(item => Label(item.Label))
 )
)
```
vs

```c#
CollectionView()
   .ItemsSource(State.Items, RenderItem())

RenderItem(MyItemType item) => Label(item.label);
```

In general use the CollectionView for performance. The Linq version is useful when you have a few items of disparate types with unique rendering needs (maybe a Settings page or something like that)

**Comment by MoDao1990217** [comment]
Start using it and plan to write an application for practice.

**Comment by juanyacovino77** [comment]
Hey fellows. I believe a great new is that David Ortinau @davidortinau (Principal Product Manager at Microsoft) just started playing around with MauiReactor by porting a .NET MAUI app to MauiReactor https://github.com/davidortinau/BalanceReactor

**Comment by adospace** [comment]
Hi everyone!
For questions, ideas, polls, etc. I've opened the discussion board here:

https://github.com/adospace/reactorui-maui/discussions

Feel free to share your use cases, questions, or code that can help others using MauiReactor.
